src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php

```diff
@@ -763,4 +763,62 @@ public function testGzipBroken()
         $this->expectException(TransportExceptionInterface::class);
         $response->getContent();
     }
+
+    public function testJsonRegexBuffer(): void
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+        $response = $client->request('GET', 'http://localhost:8057', ['buffer' => '/([a-z]+\/json)/']);
+
+        $firstContent = $response->getContent();
+        $secondContent = $response->getContent();
+
+        $this->assertSame($firstContent, $secondContent);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\HttpClient\Exception\TransportException
+     */
+    public function testFalseBuffer(): void
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+        $response = $client->request('GET', 'http://localhost:8057', ['buffer' => false]);
+
+        $firstPosts = $response->getContent();
+        $secondPosts = $response->getContent();
+    }
+
+    /**
+     * @expectedException \Symfony\Component\HttpClient\Exception\InvalidArgumentException
+     */
+    public function testInvalidRegexBuffer(): void
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+        $response = $client->request('GET', 'http://localhost:8057', ['buffer' => '/([a-z]+\/json))/']);
+
+        $firstPosts = $response->getContent();
+        $secondPosts = $response->getContent();
+    }
+
+    /**
+     * @expectedException \Symfony\Component\HttpClient\Exception\InvalidArgumentException
+     */
+    public function testPdfRegexBuffer(): void
```

Comment:
should be moved too (but I don't get what this tests?)

Reply:
Removed the test since it didn't have a meaning